Fix agent plugin build failures

Josh-Matsuoka opened the pull request:
Hi,

This patch accomplishes a few things to fix the current build failures with the agent plugin

- Updates the parent pom version to 8.1.0 to bring it in line with the current JMC version
- Updates the patch that is applied by the build scripts to bring the agent plugin poms in line with the other jmc plugins, i.e. using ${revision}${changelist} rather than explicit versions
- Updates the spotbugs excludes to exclude two non-bugs (the first is an unused return value that we don't need since the case in which it would return false is already caught by the conditions for reaching there in the first place, the second is a false positive. There are other instances of this within JMC).

aptmac commented:
Looks like there's still a couple references to `8.0.0-SNAPSHOT`, specifically in:
https://github.com/rh-jmc-team/jmc-agent-plugin/blob/17580a15d6f75fa5cc2bbea9570d85e0a5194fcb/pom.xml#L46
https://github.com/rh-jmc-team/jmc-agent-plugin/blob/17580a15d6f75fa5cc2bbea9570d85e0a5194fcb/org.openjdk.jmc.feature.console.ext.agent/pom.xml#L47
https://github.com/rh-jmc-team/jmc-agent-plugin/blob/17580a15d6f75fa5cc2bbea9570d85e0a5194fcb/org.openjdk.jmc.console.ext.agent/pom.xml#L49

Local build and format checks (after the 3 changes above) look to be good.

Josh-Matsuoka commented:
> Looks like there's still a couple references to `8.0.0-SNAPSHOT`, specifically in:
> 
> https://github.com/rh-jmc-team/jmc-agent-plugin/blob/17580a15d6f75fa5cc2bbea9570d85e0a5194fcb/pom.xml#L46
> 
> https://github.com/rh-jmc-team/jmc-agent-plugin/blob/17580a15d6f75fa5cc2bbea9570d85e0a5194fcb/org.openjdk.jmc.feature.console.ext.agent/pom.xml#L47
> 
> https://github.com/rh-jmc-team/jmc-agent-plugin/blob/17580a15d6f75fa5cc2bbea9570d85e0a5194fcb/org.openjdk.jmc.console.ext.agent/pom.xml#L49
> 
> Local build and format checks (after the 3 changes above) look to be good.

I fixed the issue with the poms that caused the previous build to fail. Those references to 8.0.0 are okay since jmc plugin versions are expected to be 8.0.0, changing those to 8.1.0 causes issues with the build. I also went ahead and cleaned up the spotbugs excludes a little bit. The new version passes all the checks.
